### PR TITLE
fix(pubsublite): correct quickstart and enable in CI

### PR DESCRIPTION
**Pull request by coryan**
The quickstart had a number of problems, starting with using the wrong
endpoint.  Pub/Sub Lite happens to be zonal, it requires a zone as a
location argument, and it uses regional endpoints.

Part of the work for #7936

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/8258)

### CODE REVIEW

File: google/cloud/pubsublite/endpoint.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#include "google/cloud/status_or.h"
+#include "google/cloud/version.h"
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace pubsublite {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+StatusOr<std::string> EndpointFromZone(std::string const& zone_id);
```
Comment (dbolduc) [inline google/cloud/pubsublite/endpoint.h:28]: I think we also want one for regions. So like:

```cc
/// A region looks like: `us-east4`
StatusOr<std::string> EndpointFromRegion(std::string const& region_id);
/// A zone looks like: `us-east4-c`
StatusOr<std::string> EndpointFromZone(std::string const& zone_id);
```

(here's why I think that). Create Topic UI:
image

I made one of each, and they are displayed like this:
image

Comment (dbolduc) [reply google/cloud/pubsublite/endpoint.h:28]: Or maybe `EndpointFromLocation`? that could get messy though.

Comment (coryan) [reply google/cloud/pubsublite/endpoint.h:35]: Thanks.  I added the new helper, but left the quickstart as-is.

File: ci/etc/integration-tests-config.ps1 [hunk 2]
```diff
@@ -12,13 +12,21 @@
...
 $env:GOOGLE_CLOUD_REGION="us-central1"
+# Some quickstart programs require a zone.
+$env:GOOGLE_CLOUD_CPP_TEST_ZONE="us-central1-a"
+
+# Enable the self-test for the sample programs. Normally the example drivers
+# require the name of the example to run as a command-line argument, with this
+# environment variable the sample drivers run all the examples.
 $env:GOOGLE_CLOUD_CPP_AUTO_RUN_EXAMPLES="yes"
+
+# A number of options to improve logging during the CI builds. The are useful
```
Comment (dbolduc) [inline ci/etc/integration-tests-config.ps1:28]: s/The/They/

File: ci/etc/integration-tests-config.sh [hunk 3]
```diff
@@ -20,13 +20,21 @@ if ((CI_ETC_INTEGRATION_TESTS_CONFIG_SH__++ != 0)); then
...
 export GOOGLE_CLOUD_REGION="us-central1"
+# Some quickstart programs require a zone.
+export GOOGLE_CLOUD_CPP_TEST_ZONE="us-central1-a"
+
+# Enable the self-test for the sample programs. Normally the example drivers
+# require the name of the example to run as a command-line argument, with this
+# environment variable the sample drivers run all the examples.
 export GOOGLE_CLOUD_CPP_AUTO_RUN_EXAMPLES="yes"
+
+# A number of options to improve logging during the CI builds. The are useful
```
Comment (dbolduc) [inline ci/etc/integration-tests-config.sh:36]: s/The/They/

**Review by dbolduc — APPROVED** [review]

File: ci/etc/integration-tests-config.ps1 (hunk 2 above)
Comment (coryan) [reply ci/etc/integration-tests-config.ps1]: Fixed.

File: ci/etc/integration-tests-config.sh (hunk 3 above)
Comment (coryan) [reply ci/etc/integration-tests-config.sh]: Fixed

**Pull request merged**